src/DiaryCard.tsx

```diff
@@ -0,0 +1,78 @@
+import {FC, useEffect, useState} from "react";
+import {Card, CardContent, makeStyles, Typography} from "@mui/material";
+import './DiaryCard.scss'
+
+type DiaryCardProps = {
+    title: string;
+    subtitle: string;
+    description: string;
+    color: string;
+}
+
+
+const DiaryCard: FC<DiaryCardProps>  = ({title, subtitle, description, color}: DiaryCardProps) => {
+
+    const [showMore, setShowMore] = useState<boolean>(false)
+
+    useEffect(() => {
+        document.title = 'Dear Diary - Home Page';
+    }, [showMore]);
+
+    return <div>
+        <Card className={"diary-card"} style={{
+            backgroundColor: color,
+        }}>
+            <CardContent>
+                <Typography
+                    className={"diary-card-title"}
+                    variant={"h5"}
+                >
+                    {title}
+                </Typography>
+                <Typography
+                    className={"diary-card-subtitle"}
+                    variant={"subtitle1"}
+                >
+                    {subtitle}
+                </Typography>
+                <Typography
+                    display={"block"}
+                    className={"diary-card-description"}
+                    variant={"body2"}
+                    style={{ wordWrap: "break-word" }}
+                >
+                    {description.length>100 && !showMore ? description.substring(0,100).concat(" ..."):description}
```

Comment:
Use a separate function to substring the string and return it

Reply:
Fixed in the latest pull request/branch  (Handle side effects)